Framework/Algorithms/src/FilterEvents.cpp

```diff
@@ -521,620 +519,181 @@ void FilterEvents::groupOutputWorkspace() {
+  double progress_step_current{0.};                          // current number of progress steps
+  const auto originalROI = m_eventWS->run().getTimeROI();
+  const bool outputUnfiltered = getProperty("OutputUnfilteredEvents");
+  for (auto const wsindex : m_targetWorkspaceIndexSet) {
     // Generate new workspace name
     bool add2output = true;
     std::stringstream wsname;
     wsname << m_outputWSNameBase << "_";
-    if (wsgroup >= 0) {
+    if (wsindex > TimeSplitter::NO_TARGET) {
       if (descriptiveNames && splitByTime) {
-        auto splitter = m_splitters[wsgroup];
-        auto startTimeInSeconds =
-            Mantid::Types::Core::DateAndTime::secondsFromDuration(splitter.start() - m_runStartTime);
-        auto stopTimeInSeconds =
-            Mantid::Types::Core::DateAndTime::secondsFromDuration(splitter.stop() - m_runStartTime);
-        wsname << startTimeInSeconds << "_" << stopTimeInSeconds;
+        TimeROI timeROI = m_timeSplitter.getTimeROI(wsindex);
+        auto timeIntervals = timeROI.toTimeIntervals();
+        for (size_t ii = 0; ii < timeIntervals.size(); ii++) {
+          auto startTimeInSeconds =
+              Mantid::Types::Core::DateAndTime::secondsFromDuration(timeIntervals[ii].start() - m_filterStartTime);
+          auto stopTimeInSeconds =
+              Mantid::Types::Core::DateAndTime::secondsFromDuration(timeIntervals[ii].stop() - m_filterStartTime);
+          wsname << startTimeInSeconds << "_" << stopTimeInSeconds;
+          if (ii < timeIntervals.size() - 1)
+            wsname << "_";
+        }
       } else if (descriptiveNames) {
-        auto infoiter = infomap.find(wsgroup);
+        auto infoiter = infomap.find(wsindex);
         if (infoiter != infomap.end()) {
           std::string name = infoiter->second;
           name = Kernel::Strings::removeSpace(name);
           wsname << name;
         } else {
-          wsname << wsgroup + delta_wsindex;
+          wsname << m_timeSplitter.getWorkspaceIndexName(wsindex, delta_wsindex);
         }
       } else {
-        wsname << wsgroup + delta_wsindex;
+        wsname << m_timeSplitter.getWorkspaceIndexName(wsindex, delta_wsindex);
       }
     } else {
       wsname << "unfiltered";
-      if (from1)
+      if (!outputUnfiltered)
         add2output = false;
     }
 
-    std::shared_ptr<EventWorkspace> optws = prototype_ws->clone();
-    // Clear Run without copying first.
-    optws->setSharedRun(Kernel::make_cow<Run>());
-    m_outputWorkspacesMap.emplace(wsgroup, optws);
+    //
+    // instantiate one of the output filtered workspaces
+    std::shared_ptr<EventWorkspace> optws = templateWorkspace->clone();
+
+    // Endow the output workspace with a TimeROI
+    TimeROI roi = this->partialROI(wsindex);
+    roi.update_or_replace_intersection(originalROI);
+
+    // discard log entries outside the ROI
+    optws->mutableRun().copyAndFilterProperties(m_eventWS->run(), roi);
+    optws->mutableRun().setTimeROI(roi);
```

Comment:
The `copyAndFilterProperties()` method already sets the TimeROI. This line is not needed.

Reply:
The problem is that `copyAndFilterProperties` is implemented in the `PropertyManager `class, while `setTimeROI `is implemented in both `PropertyManager `and `Run `classes.  So in this case:

`optws->mutableRun().setTimeROI(roi)` 

calls the `Run `version, which not only sets the `ROI`, but also calls `integrateProtonCharge `and `setDuration`.  Perhaps we need to move the `copyAndFilterProperties` method to the `Run` class?  Then, indeed, we wouldn't have to call `SetTimeROI` after calling `copyAndFilterProperties`.

Reply:
Update: the problem is solved by adding an optional boolean:

`virtual void setTimeROI(const Kernel::TimeROI &timeroi, const bool applyFilter = true);`

and calling `setTimeROI` from `PropertyManager::copyAndFilterProperties`

`this->setTimeROI(timeROI, false);`